README.md

```diff
@@ -1,14 +1,76 @@
+
+## Overall Design
+
+Our User input is contained within the repl input class. When command is called our repl input takes that command string breaks up its arguments and uses those arguments to call the proper command handler. This handlers output is then added to our replhistory. Repl hsitory adds our command output to the screen where it is displayed to users. There
+was a tradeoff between if-else nesting depth and the number of times we call modeifyOutput. We chose to have deeper nesting depth but call modeifyOutput only once. This is an aesthetic choice.
+
 # Errors/Bugs
 
+No known errors
+
 # Tests
 
+Our testing suite is contained within App.spec.ts This file houses all testing for the front end of our program and includes the following methods:
+
+on page load I see a login button: tests that program generates a login button
+on page load, i dont see the input box until login: tests that users cannot input/run commands without first logging in
+improper command input returns 'command not found': tests that the proper response is returned when no known command is input
+that Mode command changes output: tests that output switches between brief and verbose with each call of the mode command
+Load with improper args: checks that loading without inputting filename and/or header bool returns the proper error message
+Load empty csv/improper fileName: tests that loading an empty csv returns the
+Load correct load/load switch: tests that the load command + proper args properly loads a file. Also tests to make sure that load works with consecutive calls
+view without load: tests that a view command without first loading returns the proper response
+successfull view: tests that view properly displays the conents of a loaded file
+view different files: tests that only the currently loaded file is able to be viewed on screen
+search with improper args/no file loaded: makes sure the proper response is returned if search has improper args/ no file loaded
+search by index/index out of bounds: tests that searching by index returns the correct data. Also tests that indices outside the bounds of the dataset return the proper error response
+test search data with one column: tests that searching through a csv with one column works properly.
+search by column/(column doesn't exist/no headers): tests that searching by column name returns the proper response. Tests that the correct response is return when the column header does not exist. Also tests that the correct message is returned when a user tries to search by headers for a file that was stated to not have headers.
+
+\*majority of tests functions test with both brief and verbose output. All commands called on multiple datasets
```

Comment:
Does this imply that some of the tests failed the brief/verbose output? If so, which ones?